src/lib.rs

```diff
@@ -160,6 +173,72 @@ impl L402Module {
             }
         }
     }
+
+    pub async fn verify_cashu_token(&self, token: &str, amount_msat: i64) -> Result<bool, String> {
+        // Decode the token from string
+        let token_decoded = match cdk::nuts::Token::from_str(token) {
+            Ok(token) => token,
+            Err(e) => {
+                eprintln!("Failed to decode Cashu token: {}", e);
+                return Err(format!("Failed to decode Cashu token: {}", e));
+            }
+        };
+        
+        // Check if the token is valid
+        if token_decoded.proofs().is_empty() {
+            return Ok(false);
+        }
+        
+        // Calculate total token amount in millisatoshis
+        let total_amount_msat: u64 = token_decoded.proofs().iter()
+            .map(|p| u64::from(p.amount) * MSAT_PER_SAT)
+            .sum();
+        
+        // Check if the token amount is sufficient
+        if total_amount_msat < amount_msat as u64 {
+            eprintln!("Cashu token amount insufficient: {} msat (required: {} msat)", 
+                total_amount_msat, amount_msat);
+            return Ok(false);
+        }

```

Comment:
You should check that the token is of the unit you expect (sat) before you do this comparison.